src/CommonLib/DirectoryObjects/DirectoryEntryWrapper.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.DirectoryServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Principal;
+using System.Text;
+
+namespace SharpHoundCommonLib;
+
+public class DirectoryEntryWrapper : IDirectoryObject {
+    private readonly DirectoryEntry _entry;
+
+    public DirectoryEntryWrapper(DirectoryEntry entry) {
+        _entry = entry;
+    }
+    
+    public bool TryGetDistinguishedName(out string value) {
+        return TryGetProperty(LDAPProperties.DistinguishedName, out value);
+    }
+
+    private bool CheckCache(string propertyName) {
+        try {
+            if (!_entry.Properties.Contains(propertyName))
+                _entry.RefreshCache(new[] { propertyName });
+
+            return _entry.Properties.Contains(propertyName);
+        }
+        catch {
+            return false;
+        }
+    }
+
+    public bool TryGetProperty(string propertyName, out string value) {
+        value = string.Empty;
+        if (!CheckCache(propertyName)) {
+            return false;
+        }
+
+        var s = _entry.Properties[propertyName][0];
+        value = s switch {
+            string st => st,
+            int i => i.ToString(),
+            _ => null
+        };
+
+        return value != null;
+    }
+
+    public bool TryGetByteProperty(string propertyName, out byte[] value) {
+        value = Array.Empty<byte>();
+        if (!CheckCache(propertyName)) {
+            return false;
+        }
+
+        var prop = _entry.Properties[propertyName].Value;
+        if (prop is not byte[] b) return false;
+        value = b;
+        return true;
+    }
+
+    public bool TryGetArrayProperty(string propertyName, out string[] value) {
+        value = Array.Empty<string>();
+        if (!CheckCache(propertyName)) {
+            return false;
+        }
+
+        var dest = new List<string>();
+        foreach (var val in _entry.Properties[propertyName]) {
+            if (val is string s) {
+                dest.Add(s);
+            }
+        }
```

Comment:
The other methods here return false if they don't match the type (string, Byte array, etc). Should this also do some type checking?

Also, is it correct to just drop array entries that are not strings?

Reply:
This is intentional, we don't deal with non-string array types anywhere currently.